face_api_dataset/dataset.py

```diff
@@ -366,37 +372,72 @@ def transform(modalities: List[Modality]) -> List[Modality]:
                 continue
             for modality in modalities:
                 for extension in _modality_files(modality):
-                    if not (self.root / f"{number}.{extension.value}").exists():
+                    if not (self._root / f"{number}.{extension.value}").exists():
                         raise ValueError(
                             f"Can't find file '{number}.{extension.value}' "
                             f"required for {modality.name} modality"
                         )
             image_numbers.add(number)
-        self.image_numbers = sorted(list(image_numbers), key=int)
-        self._image_sizes = {}
-        self.transform = transform
+        self._image_numbers = sorted(list(image_numbers), key=int)
+        self._image_sizes: Dict[str, int] = {}
+        self._transform = transform
 
-    def __len__(self) -> int:
-        return len(self.image_numbers)
+    @property
+    def segments(self) -> Dict[str, int]:
+        """
+        Segment mapping for the dataset.
 
-    def __getitem__(self, i: int) -> dict:
-        if self.transform is None:
-            return self._get(i)
+        :type: Dict[str, int]
+        """
+        return self._segments
+
+    @property
+    def modalities(self) -> List[Modality]:
+        """
+        List of the loaded modalities.
+
+        :type: List[Modality]
+        """
+        return self._modalities
+
+    def __len__(self) -> int:
+        return len(self._image_numbers)
+
+    @overload
+    def __getitem__(self, index: int) -> dict:
+        pass
+
+    @overload
+    def __getitem__(self, index: slice) -> "FaceApiDataset":
+        pass
+
+    def __getitem__(self, i: Union[int, slice]) -> Union[dict, "FaceApiDataset"]:
```

Comment:
I'm not a big fan of else statements after a return. But dealer's choice

Reply:
Usually, I'm following this logic:
  Some exceptional situation or short-cut in logic -> no else statement.
  Choice from two (or more) possibilities -> else statement.